codeforamerica/vita-min
# Nj 116 qualifying child

## Link to pivotal/JIRA issue

https://github.com/newjersey/affordability-pm/issues/116

## Is PM acceptance required? (delete one)

Yes - don't merge until JIRA issue is accepted!

**Reminder**: merge main into this branch and get green tests before merging to main

## What was done?
- Added calculator functions to get the two categories of dependents, and added to line 13 and later relevant lines
- The new calculator functions will consider `qualifyingChild` missing from the JSON to be falsey and thus in the "other" category. (In other words, `qualifyingChild` must be both present and `true` to be a qualifying child, and any others are "other")
- No new question or review item-- all happens behind the scenes for the user
- The XML for `NumOfQualiDependChild` and `NumOfOtherDepend` already existed but was meant to be later removed. [This was discussed on the ticket](https://github.com/newjersey/affordability-pm/issues/116#issuecomment-2483766125) and so I removed that for this new logic.
- I had to change the `dependents` function in `direct_file_json_data` from `private` to public in order to be able to call it in the calculator
- Since the exemption amounts are not in the XML (just the number of dependents are), I broke our rule to populate the PDF purely from the XML and get it from the calculator-- also meaning i needed to add `calculated_fields` to the pdf filler
- Edited existing JSONs to get certain results:
  - Zeus One Dep, which is the default, is unchanged and thus has 1 "other" dependent
  - Minimal is unchanged has 0 dependents
  - Zeus Two Deps now has 1 qualifying child and 1 "other" dependent
  - Zeus Many Deps now has 10 qualifying child and 1 "other" dependent
 
## How to test?
- Try with Minimal, Zeus One Dep, Zeus Two Deps, and Zeus Many Deps to see that lines 10, 11, and 13 are correct.

## Screenshots (for visual changes)

### Minimal

https://github.com/user-attachments/assets/2a351968-9f10-463a-89ef-a35c65caeab0

---

### Zeus One Dep

https://github.com/user-attachments/assets/5008629a-4fad-4f04-a82f-c2dbd6c41728

---

### Zeus Two Deps

https://github.com/user-attachments/assets/954b3d67-7835-4439-8dc5-26946e8544d7

---

### Zeus Many Deps

https://github.com/user-attachments/assets/5d3f59d3-b98e-4ab2-9e4b-3be2016cbcc8
